lua/autorun/server/vguiCore.lua

```diff
@@ -958,6 +957,26 @@ end)
 hook.Add("PlayerLeaveVehicle", "E2VguiCloseVguiOnLeaveVehicle", function(player, vehicle)
     E2VguiCore.PlayerLeaveVehicle(player, vehicle)
 end)
+
+E2VguiCore.registerCallback("finished_loading", function()
+    -- Register callback after (re-)loading wire cores (initial loading and reloading via wire_expression2_reload)
+    registerCallback("destruct", function(self)
+        E2VguiCore.RemoveAllPanelsOfE2(self.entity.e2_vgui_core_session_id)
+
+--TODO: Check of this is served any purpose
+--         if entity:GetOwner().e2_vgui_core_default_players != nil then
+--             entity:GetOwner().e2_vgui_core_default_players[entity:EntIndex()] = nil
+--         end
+    end)
+    registerCallback("construct", function(self)
+        -- give each e2 a random session id (was originally e2EntityID)
+        -- this identifier prevents problems when reloading the same e2 (because it used the same identifier for the panels)
+        -- when reloading we tell all clients to remove the old panels (see e2 "destruct" callback above)
+        -- and then generate a new session_id (here in the e2 "construct" event) which is used to identify panels for the reloaded e2
+        -- any events from the old e2 code (panel removal/closing, button clicks, etc.) will be ignored for the new e2 session
+        self.entity.e2_vgui_core_session_id = math.floor(math.Rand(0, 1000000))
```

Comment:
Validate that there is no other e2 spawned with the same e2_vgui_core_session_id.
Although its really unlikely, adding a simple check that goes through all e2 entities should be too big of an overhead, considering this only runs when the e2 gets spawned/reloaded.

Otherwise maybe add a global list that keeps track of all used ids?

Reply:
solved via `E2VguiCore.sessionIDToE2Entity` list